UnitySDK/Assets/ML-Agents/Examples/SharedAssets/Scripts/ModelOverrider.cs

```diff
@@ -0,0 +1,58 @@
+using System;
+using UnityEngine;
+using Barracuda;
+using System.IO;
+namespace MLAgents
+{
+    /// <summary>
+    /// Utility class to allow the NNModel file for an agent to be overriden during inference.
+    /// This is useful to validate the file after training is done.
+    /// </summary>
+    public class ModelOverrider : MonoBehaviour
+    {
+        string m_AssetPath;
+        const string k_CommandLineFlag = "--mlagents-override-model";
+        
+        void OnEnable()
+        {
+            m_AssetPath = GetAssetPathFromCommandLine();
+            if (m_AssetPath != null)
+            {
+                OverrideModel();
+            }
+        }
+
+        /// <summary>
+        /// Get the asset path to use from the commandline arguments.
+        /// </summary>
+        /// <returns></returns>
+        static string GetAssetPathFromCommandLine()
+        {
+            var args = Environment.GetCommandLineArgs();
+            for (var i = 0; i < args.Length; i++)
+            {
+                if (args[i] == k_CommandLineFlag)
+                {
+                    return args[i + 1];
+                }
+            }
+
+            return null;
+        }
+        
+        /// <summary>
+        /// Load the NNModel file from the specified path, and give it to the attached agent.
+        /// </summary>
+        void OverrideModel()
+        {
+            var model = File.ReadAllBytes(m_AssetPath);
+            var asset = ScriptableObject.CreateInstance<NNModel>();
+            asset.Value = model;
+            asset.name = "Override - " + Path.GetFileName(m_AssetPath);
+            
+            var agent = GetComponent<Agent>();
```

Comment:
This will get the first component Agent in the scene. If it contains multiple Agents, this might not work. 
I think it won't work at all for WallJump but if it is only for our demo environments then this is fine.

Reply:
Yeah, will probably need some more logic with multiple brains on the the same object, or multiple agent types in the scene (e.g. soccer). I think probably specify key-value pairs on the command line, and remap them here.

Reply:
Do we ever have a GameObject with multiple agents? WallJump has 1 agent and 1 BehaviorParameters (per prefab)

Reply:
Added a comment in the summary that it will only work with scenes that have 1:1 Agent:Behavior.

Reply:
It won't get the first Agent in the scene, but the first Agent on the gameobject this is also attached to.

Reply:
I don't recall if we have multiple agents on a single gameobject